system-test/test.sh

```diff
@@ -55,17 +55,31 @@ if [[ "$VERIFY_TIME_LINE_NUMBERS" != "true" ]]; then
   npm run compile
 fi
 
+NODE_VERSION=$(node -v | cut -d. -f1 | tr -d 'v')
 node -v
 node --trace-warnings "$BENCHPATH" 10 $VERIFY_TIME_LINE_NUMBERS
 
 if [[ "$VERIFY_TIME_LINE_NUMBERS" == "true" ]]; then
-  pprof -lines -top -nodecount=2 time.pb.gz | tee $tty | \
-      grep "busyLoop.*src/busybench.js:[2-3][08-9]"
-  pprof -filefunctions -top -nodecount=2 heap.pb.gz | tee $tty | \
-      grep "busyLoop.*src/busybench.js"
+  output=$(pprof -lines -top -nodecount=2 time.pb.gz | tee $tty)
+
+  # Due to V8 changes in Node 21, the line numbers are different.
```

Comment:
Thanks for the PR. It's unfortunate that the function name is lost. Do you have any more info on this V8 change or if there is something we can do to workaround it?

It's good to see the output is the same in older node versions.

Reply:
Last time I looked at this, I did some playing around and found some weird behavior. https://github.com/google/pprof-nodejs/pull/284 somehow restores the function name

Reply:
Ah, hm. I am no v8 expert, but from some quick searches, this could be because v8's optimizer was changed/improved in version 11 (which node21 is using). Since `busyLoop` is in the "hot path", there might be some under the hood optimizations happening to it that hide it from the profiler, which adding `noop` to break it up removes.

Reply:
I'll try to dig for a way to force this, it may even require some changes to how `pprof` is being called altogether.

Reply:
I think I just proved this too, adding this native syntax yields function name in v21 (without the `noop` function)
```
function busyLoop(durationSeconds) {
  %NeverOptimizeFunction(busyLoop);
```
You have to execute the benchmark with `--allow-natives-syntax`, though.

The output:
```
Showing top 2 nodes out of 6
      flat  flat%   sum%        cum   cum%
    1805ms 68.58% 68.58%     2606ms 99.01%  (anonymous) file:/tmp/tmp.YTvrn84YUd/busybench/src/busybench.js:37
     634ms 24.09% 92.67%      634ms 24.09%  busyLoop file:/tmp/tmp.YTvrn84YUd/busybench/src/busybench.js:32'
```
interesting that there is still one trace that is anonymous, though,

Reply:
@benminer that makes sense and kind of what I figured. Thank you for looking into this. It sounds good to merge this then, I'll let @rohitpruthi-google take it over.